src/core/namespace.c

```diff
@@ -904,10 +907,22 @@ int setup_namespace(
                 if (r < 0)
                         return r;
 
-                r = dissect_image(loop_device->fd, NULL, 0, &dissected_image);
+                r = root_hash_load(root_image, &root_hash, &root_hash_size);
                 if (r < 0)
                         return r;
 
+                r = dissect_image(loop_device->fd, root_hash, root_hash_size, &dissected_image);
+                if (r < 0)
+                        return r;
+
+                r = dissected_image_decrypt(dissected_image, NULL, root_hash, root_hash_size, dissect_image_flags, &decrypted_image);
+                if (r < 0) {
+                        log_open();
+                        log_error_errno(r, "FAILED TO DECRYPT: %m");
```

Comment:
nope. this was just debugging left-over. The error message is supposed to be generated by the caller...